packages/components/src/button/README.md

```diff
@@ -218,6 +218,18 @@ If provided, renders a [Tooltip](/packages/components/src/tooltip/README.md) com
 
 -   Required: No
 
+#### `size`: `'default'` | `'compact'` | `'small'`
+
+The size of the button.
+
+-   `'default'`: For normal text-label buttons, unless it is a toggle button.
+-   `'compact'`: For toggle buttons and icon buttons.
+-   `'small'`: For icon buttons associated with more advanced or auxiliary features.
```

Comment:
@jasmussen Do these guidelines look ok?

Reply:
This part of the sheet visually explains the difference between 40 and 32:

<img width="1151" alt="Screenshot 2023-06-23 at 14 42 39" src="https://github.com/WordPress/gutenberg/assets/1204802/2d0cdb5e-a62a-4368-8881-35c2f918c1b5">

Specifically, 40px is the default to use.

32px, and compact is a good word for it, is the one to use in certain situations, notably in the case of mixing toggle buttons with buttons.

So, we could tweak "compact":

* `'compact'`: For toggle buttons, icon buttons, and buttons when used in context of either.

What do you think?